zmq.hpp

```diff
@@ -2794,6 +2794,70 @@ inline std::ostream &operator<<(std::ostream &os, const message_t &msg)
     return os << msg.str();
 }
 
+using timer_id_t = int;
+
+class timers_t
+{
+  public:
+    timers_t() : _timers(zmq_timers_new())
+    {
+        if (_timers == nullptr)
+            throw error_t();
+    }
+
+    timers_t(timers_t &other) = delete;
+
+    ~timers_t() { ZMQ_ASSERT(zmq_timers_destroy(&_timers) == 0); }
```

Comment:
Move zmq_timers_destroy out of the assert, and only check the return value in the assert.